tests/client/test_client.py

```diff
@@ -87,6 +87,35 @@ def test_graphql_errors_get_raised(patch_post):
         client.graphql("query: {}")
 
 
+@pytest.mark.parametrize("compressed", [True, False])
+def test_client_deploy(patch_post, compressed):
+    if compressed:
+        response = {
+            "data": {
+                "project": [{"id": "proj-id"}],
+                "createFlowFromCompressedString": {"id": "long-id"},
+            }
+        }
+    else:
+        response = {
+            "data": {"project": [{"id": "proj-id"}], "createFlow": {"id": "long-id"}}
+        }
+    patch_post(response)
+
+    with set_temporary_config(
+        {"cloud.graphql": "http://my-cloud.foo", "cloud.auth_token": "secret_token"}
+    ):
+        client = Client()
+    flow = prefect.Flow(name="test", storage=prefect.environments.storage.Memory())
+    flow_id = client.deploy(
```

Comment:
Can you add `with pytest.warns()` to test (and suppress) the deprecation warning?